### PR TITLE
Changed meta field names for protocol handlers.

### PR DESCRIPTION
**Pull request by dschrimpsher**
#### What does this PR do?
Changes the field names for the protocol handler metadata
#### Do you have any concerns with this PR?
no
#### How can the reviewer verify this PR?
examine the code for each PH.
#### Any background context you want to provide?
Request from UI
#### Questions:
- Have you connected this PR to the issue it resolves?
Yes
- Does the documentation need an update?
No
- Does this add new dependencies?
No
- Have you added unit or functional tests for this PR?
No

### CODE REVIEW

File: rest/networkProtocols/LoRaOpenSource.js [hunk 1]
```diff
@@ -20,18 +20,26 @@ module.exports.metaData =
...
-        fieldType: 'string',
-        displayWithQueryParameter: ''
+        name: 'username',
+        description: 'Username',
+        help: 'The username of the LoraOS admin account',
+        type: 'string',
+        label: 'Username',
+        value: '',
+        required: true,
+        placeholder: 'LoraOS Username',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/LoRaOpenSource.js]: placeholder: `e.g. 'myLoraUsername'`

File: rest/networkProtocols/LoRaOpenSource.js [hunk 2]
```diff
@@ -20,18 +20,26 @@ module.exports.metaData =
...
+        oauthQueryParameter: ''
       },
       {
-        fieldName: 'password',
-        fieldDesc: 'Password',
-        fieldHelp: 'The password of he LoraOS admin account',
-        fieldType: 'password',
-        displayWithQueryParameter: ''
+        name: 'password',
+        description: 'Password',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/LoRaOpenSource.js]: description: 'The password of he LoraOS admin account',

File: rest/networkProtocols/LoRaOpenSource.js [hunk 3]
```diff
@@ -20,18 +20,26 @@ module.exports.metaData =
...
+        help: 'The password of he LoraOS admin account',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/LoRaOpenSource.js]: can omit help for now, we may populate later

File: rest/networkProtocols/LoRaOpenSource.js [hunk 4]
```diff
@@ -20,18 +20,26 @@ module.exports.metaData =
...
+        type: 'password',
+        label: 'Password',
+        value: '',
+        required: true,
+        placeholder: 'LoraOS Password',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/LoRaOpenSource.js]: placeholder: `e.g. 'yourLoraOSPassword'`

File: rest/networkProtocols/Loriot.js [hunk 5]
```diff
@@ -17,12 +17,15 @@ module.exports = {
...
       protocolHandlerNetworkFields: [
         {
-          fieldName: 'apikey',
-          fieldDesc: 'API Key',
-          fieldHelp: 'The api key of the LoraOS admin account',
-          fieldType: 'string',
-          displayWithQueryParameter: ''
-        }
+          name: 'apikey',
+          description: 'API Key',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/Loriot.js]: description: 'The api key of the LoraOS admin account',

File: rest/networkProtocols/Loriot.js [hunk 6]
```diff
@@ -17,12 +17,15 @@ module.exports = {
...
+          help: 'The api key of the LoraOS admin account',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/Loriot.js]: can omit for now, may populate later for a help popover

File: rest/networkProtocols/Loriot.js [hunk 7]
```diff
@@ -17,12 +17,15 @@ module.exports = {
...
+          type: 'string',
+          label: 'Loriot API Key',
+          value: '',
+          required: true,
+          placeholder: 'Loriot API Key',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/Loriot.js]: "e.g. '2f70e26deaca8174b451dcc52cf68c3f...'"

File: rest/networkProtocols/TheThingsNetwork.js [hunk 8]
```diff
@@ -17,25 +17,37 @@ module.exports = {
       oauthUrl: 'https://account.thethingsnetwork.org/users/authorize',
       protocolHandlerNetworkFields: [
         {
-          fieldName: 'clientId',
-          fieldDesc: 'Client Id',
-          fieldHelp: 'The client id chosen when registering LPWan',
-          fieldType: 'string',
-          displayWithQueryParameter: ''
+          name: 'clientId',
+          description: 'Client Id',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: description: 'The client id chosen when registering LPWan'

File: rest/networkProtocols/TheThingsNetwork.js [hunk 9]
```diff
@@ -17,25 +17,37 @@ module.exports = {
...
+          help: 'The client id chosen when registering LPWan',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: can be omitted for now, at some point we can populate w info RE how to generate client id

File: rest/networkProtocols/TheThingsNetwork.js [hunk 10]
```diff
@@ -17,25 +17,37 @@ module.exports = {
...
+          type: 'string',
+          label: 'Client ID',
+          value: '',
+          required: true,
+          placeholder: 'TTN Client ID',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: placeholder: "e.g" 'your-things-client-id'"

File: rest/networkProtocols/TheThingsNetwork.js [hunk 11]
```diff
@@ -17,25 +17,37 @@ module.exports = {
...
+          help: 'The client secret provided when registering LPWan',
+          type: 'string',
+          label: 'Client Secret',
+          value: '',
+          required: true,
+          placeholder: 'TTN Client Secret',
+          oauthQueryParameter: ''
         },
         {
-          fieldName: 'accessCode',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: I personally think including code as one of the fields will be confusing to the user (even if read only). They will have no idea what it is, or how to use it. I know that @JimCAtSentosa likes it for trouble shooting purposes, but I would propose that we display code in the error message to the user, for trouble shooting purposes when there is a problem

What you you think @dschrimpsher

File: rest/networkProtocols/TheThingsNetwork.js [hunk 12]
```diff
@@ -17,25 +17,37 @@ module.exports = {
...
+          oauthQueryParameter: ''
         },
         {
-          fieldName: 'clientSecret',
-          fieldDesc: 'Client Secret',
-          fieldHelp: 'The client secret provided when registering LPWan',
-          fieldType: 'string',
-          displayWithQueryParameter: ''
+          name: 'clientSecret',
+          description: 'Client Secret',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: description: 'The client secret provided when registering LPWan',

File: rest/networkProtocols/TheThingsNetwork.js [hunk 13]
```diff
@@ -17,25 +17,37 @@ module.exports = {
...
+          help: 'The client secret provided when registering LPWan',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: can omit for now, eventually can populte w instrucitons on how to generate TTN client secret for help popover

File: rest/networkProtocols/TheThingsNetwork.js [hunk 14]
```diff
@@ -17,25 +17,37 @@ module.exports = {
...
+          type: 'string',
+          label: 'Client Secret',
+          value: '',
+          required: true,
+          placeholder: 'TTN Client Secret',
```
Comment (SteveAtSentosa) [inline rest/networkProtocols/TheThingsNetwork.js]: placeholder: "e.g. 'ZDTXlyIatAHYPDBOXnJUAVnwjqkMsrR...''"

**Comment by jcampanell-cablelabs** [comment]
Regarding the displayWithQueryParameter field:
- First, maybe a bad name - "oauthQueryParameter"?
- I would display not only for troubleshooting, but to show that the OAuth process actually happened (or not, in case of system or user error).

But whatever you guys think.  It's just, like, my opinion, man.

**Comment by SteveAtSentosa** [comment]
How about we display text that says "Things Network Authentication Succeeded" upon success, and then show code in error message if there is a problem

**Comment by jcampanell-cablelabs** [comment]
Sure.  I just think there should be some indication of whether or not the OAuth process happened.

**Comment by dmalas** [comment]
I agree! It needs to be clear what failed, too.

**Comment by dschrimpsher** [comment]
@jcampanell-cablelabs ?
screen shot 2018-07-31 at 11 46 24 am

**Comment by jcampanell-cablelabs** [comment]
Sorry. Sometimes I just start typing without regard to my "persona".  The AtSentosa is a very old account Steve set up for me for an old project I did with him.  And for some reason, it came up as the default when I started saving lpwanserver to github.  jcampanell-cablelabs is my official cablelabs account.

**Comment by dschrimpsher** [comment]
No what I am confused about is 

Comment #1:  So maybe a better name is just "oauthQueryParameter".
Comment #2: First, maybe a bad name - "oauthQueryParameter"?

**Comment by jcampanell-cablelabs** [comment]
Oh, sorry, didn't realize that first comment was already in this thread.  In both cases, proposing he same change to oauthQueryParameter.  At least I'm consistent. ;)

File: rest/networkProtocols/TheThingsNetwork.js (hunk 11 above)
Comment (dschrimpsher) [reply rest/networkProtocols/TheThingsNetwork.js]: I just provide it.  I leave it up to you "UI people" to figure out what to show.  :-)

**Comment by dschrimpsher** [comment]
It is thats what I am confused by.  I thought the second one was to change it back

**Comment by dschrimpsher** [comment]
Should we have a "readonly" property.  Since the access-code is read only for instance.

**Comment by SteveAtSentosa** [comment]
ship it

**Pull request merged**